src/test/java/org/broadinstitute/hellbender/utils/downsampling/ReadsDownsamplingIteratorUnitTest.java

```diff
@@ -0,0 +1,127 @@
+package org.broadinstitute.hellbender.utils.downsampling;
+
+import htsjdk.samtools.TextCigarCodec;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class ReadsDownsamplingIteratorUnitTest extends BaseTest {
```

Comment:
while you're here, add tests for error conditions: calling `next()` when there are no items and calling `remove`